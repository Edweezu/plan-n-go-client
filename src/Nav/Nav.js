import React from 'react'
import { Link } from 'react-router-dom'
import TokenService from '../services/token-service'
import TripsContext from '../TripsContext';
import IdleService from '../services/idle-service'

class Nav extends React.Component {
    
        static contextType = TripsContext

        handleLogoutClick = () => {
            this.context.loggedOut()
            TokenService.clearAuthToken()
            TokenService.clearCallbackBeforeExpiry()
            IdleService.unRegisterIdleResets()
        }

        renderLogoutLink() {
            return (
              <div className='Header-in'>
                <h2>
                  <Link to='/dashboard'><i className="fas fa-home"></i>
                      Plan-n-Go
                  </Link>
                </h2>
                <div className='Header-links'>
                  <Link
                    onClick={this.handleLogoutClick}
                    to='/'>
                    Logout
                  </Link>
                </div>          
              </div>
            )
          }
        
          renderLoginLink() {
            return (
              <div className='Header-not-in'>
                <h2>
                  <Link to ='/'><i className="fas fa-home"></i>
                      Plan-n-Go
                  </Link>
                </h2>
                <div className='Header-links'>
                  <Link
                    to='/register'>
                    Create an Account
                  </Link>
                  <span className='Header-space'>  
                  </span>
                  <Link
                    to='/login'>
                    Log in
                  </Link>
                </div>   
              </div>
            )
          }

        
    render () {
        // console.log('tokennn', TokenService.hasAuthToken())
        // console.log('contexttt', this.context)


        // const { login } = this.context
        return (
            <nav className='nav'>
                {TokenService.hasAuthToken()
                    ? this.renderLogoutLink()
                    : this.renderLoginLink()
                }
            </nav>   
        )
    }
}

export default Nav;