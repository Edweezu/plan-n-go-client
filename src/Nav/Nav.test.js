import React from 'react'
import toJson from 'enzyme-to-json'
import { shallow } from 'enzyme'
import Nav from './Nav'


describe('Nav component', () => {
    it('renders the given component', () => {
        const wrapper = shallow(<Nav />)
        expect(toJson(wrapper)).toMatchSnapshot()
    })
})