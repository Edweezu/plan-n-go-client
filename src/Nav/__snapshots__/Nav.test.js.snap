// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Nav component renders the given component 1`] = `
<nav
  className="nav"
  id="navbar"
>
  <div
    className="Header-not-in"
  >
    <h3>
      <Link
        to="/"
      >
        <i
          className="fas fa-plane"
        />
        Plan-n-Go
      </Link>
    </h3>
    <div
      className="Header-links"
    >
      <Link
        to="/register"
      >
        Create an Account
      </Link>
      <span
        className="Header-space"
      />
      <Link
        to="/login"
      >
        Login
      </Link>
    </div>
  </div>
</nav>
`;
